Assignment5/miguelangel43-21A355/assignment_5.py

```diff
@@ -0,0 +1,268 @@
+
+Additional information can be found in https://kgtk.readthedocs.io/
+
+## Simple graph statistics
+
+Let's calculate first some statistics about the KG. Count the number of instances:
+"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#       --match '(x)-[]->()'
+#       --return 'count(x) as N'
+# """)
+
+"""Now, count the number of distinct properties: 
+
+"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#       --match '()-[x]->()'
+#       --return 'count(distinct x.label) as N'
+# """)
+
+"""Now, let's count the frequency of those properties. That is, how many instances we can find with each property"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#       --match '(x)-[r]->(y)' \
+#       --return 'r.label as property, count(x) as frequency'
+# """)
+
+"""## Simple queries
+Some of these queries are simple and will run in the Wikidata endpoint. 
+Try both of them using SPARQL and Kypher
+"""
+
+# Which actors has Schwarzenegger worked with throughout his career? (Print also the movie)
+
+# (in SPARQL)
+# SELECT ?actor ?movie WHERE {
+#   ?movie wdt:P161 wd:Q2685.
+#   ?movie wdt:P161 ?actor.
+#   FILTER (?actor != wd:Q2685)
+# }
+
+# In Kypher:
+kgtk("""
+    query -i all
+      --match '(:Q2685)<-[:P161]-(movie)-[:P161]->(z)'
+      --where 'z != "Q2685"'
+      --return 'z, movie'
+      / add-labels
+""")
+
+# How many awards does Schwarzenegger have?
+
+# SPARQL:
+# SELECT (count(?awards) as ?c_awards) WHERE {
+#   wd:Q2685 wdt:P166 ?awards. 
+# }
+
+# Kypher:
+kgtk("""
+    query -i all
+      --match '(:Q2685)-[:P166]->(awards)'
+      --return 'count(awards) as awards'
+""")
+
+# Retrieve at least two members of Schwarzenegger's political party. Make sure only persons are returned
+# SPARQL:
+# SELECT ?chairman WHERE {
+#   wd:Q2685 wdt:P102 ?political_party.
+#   OPTIONAL { ?chairman wdt:P102 ?political_party. }
+# }
+# LIMIT 5
+
+# Kypher:
+kgtk("""
+    query -i all
+      --match '(:Q2685)-[:P102]->(party)<-[:P102]-(chairman)-[:P31]->(:Q5)'
+      --return 'distinct chairman'
+""")
+
+# What are the properties that describe an artist?
+
+# In theory this one is heavy on Wikidata
+
+# SPARQL: 
+# SELECT DISTINCT ?prop WHERE {
+#    ?x rdfs:subClassOf* wd:Q483501.
+#    ?element wdt:P106 ?x.
+#    ?element ?prop ?p.
+#  }
+
+# In Kypher:
+kgtk("""
+    query -i all 
+        --match '()<-[l {label: property}]-(x)-[:P106]->(y)-[:P279*]->(:Q483501)' 
+        --return 'distinct property'
+    / add-labels
+""")
+
+# And a film director?
+# SPARQL: 
+
+
+# In Kypher:
+kgtk("""
+    query -i all
+        --match '()<-[l {label: prop}]-(x)-[:P106]->(y)-[:P279star]->(:Q2526255)'
```

Comment:
Why here the P279 property is named differently? It looks like you have not tried these queries.